cmd/snap-seccomp/versioninfo.go

```diff
@@ -0,0 +1,503 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"os"
+
+	"github.com/mvo5/libseccomp-golang"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+// Generated using arch-syscall-dump test tool from libseccomp tree, git
+// revision 584ca7a5e69d87a4c2c4e4c07ce8415fa59e1351.
```

Comment:
did you run that tool for all our supported archs and took the union?  can't we build the tool in our tests (from a most recent stable branch) and check the output hasn't changed?

Reply:
Yes, I did it for all arches supported by libseccomp.

We could have a test that builds libseccomp from master, runs the tool and verifies we accounted for all of the syscalls.

Reply:
Note that libseccomp has weird notion of system calls. It doesn't differentiate architecture specific system calls from current architecture system calls. It's just a set, equal on all architectures.

Reply:
Added a test in https://github.com/snapcore/snapd/pull/6592/commits/70a0679151a24156e7af3c867f4d688faeddefc1 Had to move the syscalls into a separate package, hopefully that's not too controversial.

Reply:
Aha, nice! Yeah, this is what I had hoped for \o/